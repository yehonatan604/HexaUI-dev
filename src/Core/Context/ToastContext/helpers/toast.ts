import { TToastVariants } from "../../../../UI/Components/Core/Toast/types/TToastVariants";

let toastInstance: TToastVariants = {};

export const setToastInstance = (instance: TToastVariants) => {
    toastInstance = instance;
};

export const toast = {
    info: (message: string) => toastInstance?.info?.(message),
    success: (message: string) => toastInstance?.success?.(message),
    error: (message: string) => toastInstance?.error?.(message),
    warning: (message: string) => toastInstance?.warning?.(message),
};
