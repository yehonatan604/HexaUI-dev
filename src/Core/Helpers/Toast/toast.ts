import { TToast } from "../../../Data/Types/TToast";

let toastInstance: TToast = {};

export const setToastInstance = (instance: TToast) => {
    toastInstance = instance;
};

export const toast = {
    info: (message: string) => toastInstance?.info?.(message),
    success: (message: string) => toastInstance?.success?.(message),
    error: (message: string) => toastInstance?.error?.(message),
    warning: (message: string) => toastInstance?.warning?.(message),
};
