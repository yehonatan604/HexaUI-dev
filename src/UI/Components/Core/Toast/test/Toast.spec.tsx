import { render, screen, waitFor, fireEvent } from "@testing-library/react";
import { toast } from "../../../../../Core/Context/ToastContext/helpers/toast";
import ToastProvider from "../../../../../Core/Context/ToastContext/providers/Toast.provider";
import { describe, beforeEach, afterEach, it, expect, vi } from "vitest";

// Mock the Toast component
vi.mock("../../../../UI/Components/Core/Toast/Toast", () => ({
  __esModule: true,
  default: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
}));

// Test component to trigger toast via button click
const TestComponent = () => (
  <ToastProvider>
    <div>Test App</div>
    <button onClick={() => toast.info("This is an info message")}>Show Info Toast</button>
    <button onClick={() => toast.success("This is a success message")}>
      Show Success Toast
    </button>
  </ToastProvider>
);

describe("Toast System", () => {
  // Create the #alert-container for the portal
  beforeEach(() => {
    const portalRoot = document.createElement("div");
    portalRoot.setAttribute("id", "alert-container");
    document.body.appendChild(portalRoot);
  });

  afterEach(() => {
    const portalRoot = document.getElementById("alert-container");
    if (portalRoot) {
      document.body.removeChild(portalRoot);
    }
  });

  it("should display an info toast using a button click and match the snapshot", async () => {
    const { asFragment } = render(<TestComponent />);

    // Simulate clicking the button to trigger the toast
    fireEvent.click(screen.getByText("Show Info Toast"));

    // The toast should appear in the alert container
    await waitFor(() =>
      expect(screen.getByText("This is an info message")).toBeInTheDocument()
    );

    // Check that the toast has the correct title
    expect(screen.getByText("Info")).toBeInTheDocument();

    // Snapshot the DOM structure after the toast is rendered
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display a success toast using a button click and match the snapshot", async () => {
    const { asFragment } = render(<TestComponent />);

    // Simulate clicking the button to trigger the toast
    fireEvent.click(screen.getByText("Show Success Toast"));

    // The toast should appear in the alert container
    await waitFor(() =>
      expect(screen.getByText("This is a success message")).toBeInTheDocument()
    );

    // Check that the toast has the correct title
    expect(screen.getByText("Success")).toBeInTheDocument();

    // Snapshot the DOM structure after the toast is rendered
    expect(asFragment()).toMatchSnapshot();
  });
});
