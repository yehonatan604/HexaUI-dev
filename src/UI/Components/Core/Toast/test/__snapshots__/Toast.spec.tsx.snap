// Vitest Snapshot v1

exports[`Toast System > should display a success toast using a button click and match the snapshot 1`] = `
<DocumentFragment>
  <div>
    Test App
  </div>
  <button>
    Show Info Toast
  </button>
  <button>
    Show Success Toast
  </button>
  <div
    id="alert-container"
  >
    <div
      id="toast-qihox1nex"
    />
  </div>
</DocumentFragment>
`;

exports[`Toast System > should display an info toast using a button click and match the snapshot 1`] = `
<DocumentFragment>
  <div>
    Test App
  </div>
  <button>
    Show Info Toast
  </button>
  <button>
    Show Success Toast
  </button>
  <div
    id="alert-container"
  >
    <div
      id="toast-n4pgftrw3z"
    />
  </div>
</DocumentFragment>
`;
